tests/phpunit/tests/rest-api/rest-posts-controller.php

```diff
@@ -1754,6 +1754,51 @@ public function test_get_items_all_post_formats() {
 		$this->assertSame( $formats, $data['schema']['properties']['format']['enum'] );
 	}
 
+	/**
+	 * @ticket 55593
+	 */
+	public function test_get_items_parent_ids_primed() {
+		$parent_id1 = $this->factory->post->create( array( 'post_title' => 'Test Post with attachment' ) );
+		$parent_id2 = $this->factory->post->create( array( 'post_title' => 'Test Post with attachment' ) );
+		$parent_ids = array( $parent_id2, $parent_id1 );
+
+		$this->factory->attachment->create_object(
+			DIR_TESTDATA . '/images/canola.jpg',
+			$parent_id1,
+			array(
+				'post_mime_type' => 'image/jpeg',
+				'post_excerpt'   => 'A sample caption',
+			)
+		);
+
+		$this->factory->attachment->create_object(
+			DIR_TESTDATA . '/images/canola.jpg',
+			$parent_id2,
+			array(
+				'post_mime_type' => 'image/jpeg',
+				'post_excerpt'   => 'A sample caption',
+			)
+		);
+
+		// Clean parents ids from cache.
+		wp_cache_delete_multiple( $parent_ids, 'posts' );
```

Comment:
Is this needed?

Reply:
Not if you use the existing fixtures per my suggestion above. The cache is flushed between each test & `wp_insert_post()` doesn't run `get_post()` on the parent as best I can tell.

If you really need to create the parents in this test then yes. `wp_insert_post()` does call `get_post()` on the new post. TIL.

https://github.com/WordPress/wordpress-develop/blob/81eb65fd9eb1a42efc657225d2ba1bce82406278/src/wp-includes/post.php#L4545

Reply:
If we create the posts, indeed is needed.
Per @peterwilsoncc suggestion, I tested using an existing fixture ( we don't really need to create new posts), but still, we need to clean the cache, since those parents' ids are already present in the cache.
I would have thought the same, that cache is flushed between tests, but maybe creating the attachment and assigning it to a parent is the guilty here, so we still need to delete the cache for the test's sake.